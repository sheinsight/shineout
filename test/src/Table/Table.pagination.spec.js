import React from 'react'
import { mount } from 'enzyme'
import { Table } from 'shineout'
import Render from 'react-test-renderer'

/* global SO_PREFIX */
const columns = [
  {
    id: 'id',
    render: 'id',
  },
  {
    title: 'Name',
    render: 'name',
  },
]

const data = [{ id: 0, name: 'Jack' }, { id: 1, name: 'Lucy' }, { id: 2, name: 'Tom' }, { id: 3, name: 'Jerry' }]
function createTable(props) {
  return <Table keygen="id" columns={columns} data={data} {...props} />
}
describe('Table[pagination]', () => {
  test('should render correctly', () => {
    const wrapper = Render.create(
      createTable({
        pagination: {
          current: 1,
          pageSize: 1,
        },
      })
    ).toJSON()
    expect(wrapper).toMatchSnapshot()
  })
  test('should use pageSize when defaultPageSize and pageSize are both specified', () => {
    const wrapper = mount(createTable({ pagination: { pageSize: 1 } }))
    expect(wrapper.find(`tr.${SO_PREFIX}-table-normal`)).toHaveLength(1)
  })
  test('repaginates when pageSize change', () => {
    const wrapper = mount(
      createTable({
        pagination: {
          pageSize: data.length,
        },
      })
    )
    expect(wrapper.find(`tr.${SO_PREFIX}-table-normal`)).toHaveLength(data.length)
    wrapper.setProps({ pagination: { pageSize: 1 } })
    expect(wrapper.find(`tr.${SO_PREFIX}-table-normal`)).toHaveLength(1)
  })
  test('should call onChange', () => {
    const handleChange = jest.fn()
    const wrapper = mount(
      createTable({
        pagination: {
          pageSize: 1,
          onChange: handleChange,
        },
      })
    )
    wrapper
      .find('ShineoutPaginationNext')
      .find('a')
      .prop('onClick')()
    expect(handleChange.mock.calls[0][0]).toBe(2)
  })
  test('should support specify the position of pagination', () => {
    const positions = ['left', 'center', 'right']
    positions.forEach(pos => {
      const wrapper = mount(
        createTable({
          pagination: {
            align: pos,
          },
        })
      )
      expect(wrapper.find(`.${SO_PREFIX}-pagination-${pos}`)).toHaveLength(1)
    })
  })
})
