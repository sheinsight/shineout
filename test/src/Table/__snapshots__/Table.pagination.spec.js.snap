// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table[pagination] should render correctly 1`] = `
Array [
  <div
    className="so-table so-table-default so-table-hover so-table-vertical-top"
    style={Object {}}
  >
    <div
      className="so-table-head so-table-simple-head"
      style={
        Object {
          "overflowY": "hidden",
        }
      }
    >
      <table
        className=""
        style={
          Object {
            "width": undefined,
          }
        }
      >
        <colgroup>
          <col
            style={
              Object {
                "width": undefined,
              }
            }
          />
          <col
            style={
              Object {
                "width": undefined,
              }
            }
          />
        </colgroup>
        <thead>
          <tr>
            <th
              className="so-table-center so-table-condensed"
              style={
                Object {
                  "padding": 0,
                }
              }
            />
            <th
              className=""
              rowSpan={1}
            >
              <div
                className=""
              >
                <span>
                  Name
                </span>
              </div>
            </th>
          </tr>
        </thead>
      </table>
    </div>
    <div
      className="so-table-simple-body"
      onScroll={[Function]}
    >
      <table
        className=""
        style={
          Object {
            "width": undefined,
          }
        }
      >
        <colgroup>
          <col
            style={
              Object {
                "width": undefined,
              }
            }
          />
          <col
            style={
              Object {
                "width": undefined,
              }
            }
          />
        </colgroup>
        <tbody>
          <tr
            className="so-table-normal"
            onClick={[Function]}
          >
            <td
              className=""
              colSpan={1}
              style={Object {}}
            >
              0
            </td>
            <td
              className=""
              colSpan={1}
              style={Object {}}
            >
              Jack
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>,
  <div
    className="so-pagination so-pagination-default"
    style={Object {}}
  >
    <div
      className="so-pagination-links so-pagination-section so-pagination-section-ltr"
    >
      <a
        className="so-pagination-item so-pagination-item-ltr so-pagination-arrow"
        disabled={true}
        onClick={[Function]}
      >
        <svg
          viewBox="0 0 1024 1024"
        >
          <path
            d="M304.905 561.68c-11.785 0-23.57-4.496-32.562-13.488-17.984-17.983-17.984-47.139 0-65.122l402.522-402.522c17.982-17.983 47.14-17.983 65.122 0 17.984 17.984 17.984 47.14 0 65.123l-402.521 402.521c-8.992 8.993-20.776 13.488-32.562 13.488z"
          />
          <path
            d="M707.426 964.201c-11.785 0-23.57-4.496-32.561-13.488l-402.523-402.522c-17.984-17.983-17.984-47.139 0-65.122 17.983-17.983 47.14-17.983 65.123 0l402.521 402.522c17.984 17.983 17.984 47.139 0 65.122-8.991 8.992-20.776 13.488-32.561 13.488z"
          />
        </svg>
      </a>
      <a
        className="so-pagination-item so-pagination-item-ltr so-pagination-current"
        disabled={true}
        onClick={[Function]}
      >
        1
      </a>
      <a
        className="so-pagination-item so-pagination-item-ltr"
        disabled={false}
        onClick={[Function]}
      >
        2
      </a>
      <a
        className="so-pagination-item so-pagination-item-ltr"
        disabled={false}
        onClick={[Function]}
      >
        3
      </a>
      <a
        className="so-pagination-item so-pagination-item-ltr"
        disabled={false}
        onClick={[Function]}
      >
        4
      </a>
      <a
        className="so-pagination-item so-pagination-item-ltr so-pagination-arrow"
        disabled={false}
        onClick={[Function]}
      >
        <svg
          viewBox="0 0 1024 1024"
        >
          <path
            d="M728.76 561.68c-11.785 0-23.57-4.496-32.561-13.488l-402.523-402.521c-17.984-17.984-17.984-47.14 0-65.123 17.983-17.983 47.14-17.983 65.123 0l402.521 402.522c17.984 17.984 17.984 47.14 0 65.122-8.991 8.993-20.776 13.488-32.561 13.488z"
          />
          <path
            d="M326.238 964.202c-11.785 0-23.57-4.496-32.562-13.488-17.984-17.983-17.984-47.139 0-65.122l402.522-402.522c17.982-17.983 47.14-17.983 65.122 0 17.984 17.984 17.984 47.14 0 65.122l-402.521 402.522c-8.992 8.992-20.776 13.488-32.562 13.488z"
          />
        </svg>
      </a>
    </div>
  </div>,
]
`;
